src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -335,8 +336,8 @@ bool CartGridStrategy::handleGcode(Gcode *gcode)
             // first wait for an empty queue i.e. no moves left
             THEKERNEL->conveyor->wait_for_idle();
 
-            int n = gcode->has_letter('I') ? gcode->get_value('I') : configured_grid_x_size;
-            int m = gcode->has_letter('J') ? gcode->get_value('J') : configured_grid_y_size;
+            int n = gcode->has_letter('I') ? gcode->get_value('I') : (gcode->has_letter('J') ? gcode->get_value('J') : configured_grid_x_size);
```

Comment:
This is some of the worst code I have seen in a while! it is totally obfuscated and ugly.